Nodejs/Product/Nodejs/Workspace/NpmActionProviderFactory.cs

```diff
@@ -0,0 +1,168 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.NodejsTools.Npm;
+using Microsoft.NodejsTools.NpmUI;
+using Microsoft.VisualStudio.Threading;
+using Microsoft.VisualStudio.Workspace;
+using Microsoft.VisualStudio.Workspace.Extensions.VS;
+using Microsoft.VisualStudioTools.Project;
+
+namespace Microsoft.NodejsTools.Workspace
+{
+    [ExportFileContextActionProvider(ProviderType, Guids.PackageJsonContextTypeString)]
+    public class NpmActionProviderFactory : IWorkspaceProviderFactory<IFileContextActionProvider>
+    {
+        private const string ProviderType = "{971EA159-8F0F-44FF-A279-B11DF1D7F319}";
+
+        [Import]
+        private OutputPaneWrapper OutputPane { get; set; }
+
+        public IFileContextActionProvider CreateProvider(IWorkspace workspaceContext)
+        {
+            return new FileContextActionProvider(workspaceContext, OutputPane);
+        }
+
+        private class FileContextActionProvider : IFileContextActionProvider
+        {
+            public readonly IWorkspace workspaceContext;
+            public readonly OutputPaneWrapper outputPane;
+
+            public FileContextActionProvider(IWorkspace workspaceContext, OutputPaneWrapper outputPane)
+            {
+                this.workspaceContext = workspaceContext;
+                this.outputPane = outputPane;
+            }
+
+            public async Task<IReadOnlyList<IFileContextAction>> GetActionsAsync(string filePath, FileContext fileContext, CancellationToken cancellationToken)
+            {
+                await this.workspaceContext.JTF.SwitchToMainThreadAsync();
+                this.outputPane.ShowWindow();
```

Comment:
Why is this called here?

Reply:
to ensure the output pane is initialized... then we no longer need to switch to the UI thread when invoking any commands.